code/modules/forensics/forensic_data.dm

```diff
@@ -64,3 +69,96 @@
 		data_copy.time_end = src.time_end
 		REMOVE_FLAG(data_copy.flags, FORENSIC_USED)
 		return data_copy
+
+/datum/forensic_data/adminprint
+	var/datum/forensic_id/clientKey = null
+
+	New(var/datum/forensic_id/clientKey)
+		..()
+		src.clientKey = clientKey
+
+	get_text()
+		var/mins_start = src.get_minutes_since(TIME, TRUE)
+		var/mins_end = src.get_minutes_since(TIME)
+		if(mins_start == mins_end)
+			return src.clientKey.id + SPAN_SUBTLE(" ([mins_end] mins ago)")
+		return src.clientKey.id + SPAN_SUBTLE(" ([mins_end] to [mins_start] mins ago)")
+
+	get_copy()
+		var/datum/forensic_data/adminprint/data_copy = new(src.clientKey)
+		data_copy.time_start = src.time_start
+		data_copy.time_end = src.time_end
+		REMOVE_FLAG(data_copy.flags, FORENSIC_USED)
+		return data_copy
+
+/datum/forensic_data/fingerprint
+	var/datum/forensic_id/print = null
+	var/datum/forensic_id/fibers = null
+	var/datum/forensic_id/print_mask = null
+
+	New(var/datum/forensic_id/print, var/datum/forensic_id/fibers, var/datum/forensic_id/print_mask, var/flags = 0)
+		..()
+		src.print = print
+		src.fibers = fibers
+		src.print_mask = print_mask
+		src.flags = flags
+
+	get_text(var/datum/forensic_scan/scan)
+		var/fibers_text = SPAN_SUBTLE(src.fibers?.id)
+		if(scan.has_effect("effect_silver_nitrate"))
+			// Silver nitrate was applied. Show partial fingerprints.
+			var/fprint_text = get_masked_print()
+			if(fprint_text && fibers_text && src.print_mask)
+				return "([fprint_text]) [fibers_text]"
+			return fprint_text + fibers_text
+		else if(src.print_mask?.id == FORENSIC_GLOVE_MASK_FINGERLESS)
+			return src.print?.id
+		else if(src.fibers && startswith(src.fibers.id, "latex rubber"))
+			return "([get_masked_print()])"
+
+			// Not including the fibers for now because they were taking up too much room in the forensic report.
+			//var/fprint_text = get_masked_print()
+			// if(fprint_text && fibers_text)
+				// return "[fprint_text] ([fibers_text])"
+			// return fprint_text + fibers_text
```

Comment:
```suggestion
```
Either uncomment the code or just remove it, we try to avoid leaving blocks of code in comments